src/nextjournal/clerk/viewer.cljc

```diff
@@ -629,19 +620,28 @@
     {:nextjournal/render-opts (-> cell (select-keys [:loc]) (assoc :id (processed-block-id (str id "-code"))))}
     (dissoc cell ::doc :result)))
 
+(defn fragment-tree-seq [{:as cell :keys [result]}]
+  (if-some [fgmt (-> result (get-safe :nextjournal/value) (get-safe :nextjournal.clerk/fragment))]
+    (mapcat (fn [r]
+              (fragment-tree-seq
+               (-> cell
+                   (assoc ::fragment-item? true)
```

Comment:
I'd much prefer not to pass around this `::fragment-item?` bool, but the _first_ element of a fragment is undistinguishable from ordinary results in this setting (others might be recognised by their paths). The flag is still needed in the frontend to apply some styling to make md results look like the rest of Clerk's prose. Maybe we can rethink how to fix the styling altogether in a next step?